presto-spark-base/src/main/java/com/facebook/presto/spark/execution/LocalTaskFactory.java

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.airlift.concurrent.BoundedExecutor;
+import com.facebook.airlift.http.client.HttpClient;
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.presto.Session;
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.execution.TaskInfo;
+import com.facebook.presto.execution.TaskSource;
+import com.facebook.presto.execution.buffer.OutputBuffers;
+import com.facebook.presto.execution.scheduler.TableWriteInfo;
+import com.facebook.presto.metadata.InternalNode;
+import com.facebook.presto.server.TaskUpdateRequest;
+import com.facebook.presto.server.remotetask.RemoteTaskStats;
+import com.facebook.presto.sql.planner.PlanFragment;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+
+public class LocalTaskFactory
+{
+    private final HttpClient httpClient;
+    private final RemoteTaskStats stats;
+    private final ExecutorService coreExecutor;
+    private final Executor executor;
+    private final ScheduledExecutorService updateScheduledExecutor;
+    private final JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec;
+    private final JsonCodec<PlanFragment> planFragmentCodec;
+    private final JsonCodec<TaskInfo> taskInfoCodec;
+
+    @Inject
+    public LocalTaskFactory(
+            @ForWorker HttpClient httpClient,
+            RemoteTaskStats stats,
+            JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec,
+            JsonCodec<PlanFragment> planFragmentCodec,
+            JsonCodec<TaskInfo> taskInfoCodec)
+    {
+        this.httpClient = httpClient;
+        this.stats = stats;
+        this.taskUpdateRequestCodec = taskUpdateRequestCodec;
+        this.planFragmentCodec = planFragmentCodec;
+        this.taskInfoCodec = taskInfoCodec;
+
+        this.coreExecutor = newCachedThreadPool(daemonThreadsNamed("local-task-callback-%s"));
+        this.executor = new BoundedExecutor(coreExecutor, 1000);
+        this.updateScheduledExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("task-info-scheduler-%s"));
+    }
+
+    public LocalTask createLocalTask(
+            Session session,
+            TaskId taskId,
+            InternalNode node,
+            URI location,
+            PlanFragment fragment,
+            List<TaskSource> sources,
+            OutputBuffers outputBuffers,
```

Comment:
We might not need a buffer to be passed in since no large results will ever be sent back. Maybe create inside would make more sense?

Reply:
we probably do even need a buffer....

Reply:
Do? or do not? I'm confused

Reply:
Currently it's used when create Initial TaskInfo (createInitialTask). I assume I can use a dummy outputBuffers in there?